src/core/src/core_logic/PatchInstaller.py

```diff
@@ -176,14 +177,49 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
         self.last_still_needed_packages = all_packages
         self.last_still_needed_package_versions = all_package_versions
 
+        stopwatch_for_batch_install_process = Stopwatch(self.env_layer, self.telemetry_writer, self.composite_logger)
```

Comment:
Repeating from what is added in the overall comment for this PR. Ensure there are minimal changes to the existing logic of sequential patching and that the code flow within parallel patching is similar to sequential (even in terms of code structure and sectioning).

A way to do that, is copy install_updates() fully into install_packages_in_batches() and then edit install_packages_in_batches() to add batching to it.
Then, just add a function call to install_packages_in_batches() in install_updates()

I'm guessing, after this, we might discover some common code that can be extracted out and re-used between these 2 functions

Reply:
The issue here and the reason behind asking for retaining the existing code as is was:

- to ensure the current flow is unaffected
- some of the changes here may affect the current code flow for single/sequential patching. 
- makes it easier to understand the code flow and also the difference between parallel and sequential.
The way the code is structured right now, 
- Ensuring there are smaller functions, again making it easier to follow.